google-cloud-core/src/main/java/com/google/cloud/Timestamp.java

```diff
@@ -43,12 +47,32 @@
       new Timestamp(253402300799L, (int) TimeUnit.SECONDS.toNanos(1) - 1);
 
   /** Regexp to split timestamps into date-hour-minute-second and fractional second components. */
-  private static final Pattern FORMAT_REGEXP = Pattern.compile("([^\\.]*)(\\.\\d{0,9})?Z");
+  private static final Pattern FORMAT_REGEXP = Pattern.compile("([^.]*)(\\.\\d{0,9})?Z");
+  private static final String DATE_GROUP = "(\\d{4}-\\d{1,2}-\\d{1,2})";
```

Comment:
why do you want to allow single digit months,dates and the various time parts below?
According to the RFC they have to be 2 digits
https://www.ietf.org/rfc/rfc3339.txt
https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html#ISO_LOCAL_DATE